src/Compilers/Core/Portable/Collections/ImmutableArrayExtensions.cs

```diff
@@ -448,6 +448,11 @@ internal static ImmutableArray<T> Concat<T>(this ImmutableArray<T> first, Immuta
             return first.AddRange(second);
         }
 
+        internal static ImmutableArray<T> Concat<T>(this ImmutableArray<T> first, T second)
+        {
+            return first.Add(second);
+        }
+
```

Comment:
Side effect of refactoring over time.  You are right that we can just inline this directly.